Net/Gearman/Connection.php

```diff
@@ -476,11 +476,21 @@ public function close()
      */
     public function isConnected()
     {
+        if ($this->socket === null) {
+            // no socket established yet
+            return false;
+        }
+
+        // PHP 8+ returns Socket object instead of resource
+        if ($this->socket instanceof \Socket) {
+            return true;
+        }
+
         // HHVM returns stream. PHP 5.x returns socket
         $type = strtolower(get_resource_type($this->socket));
-        return (is_null($this->socket) !== true &&
-                is_resource($this->socket) === true &&
-                ($type == 'socket' || $type == "stream"));
+
+        return is_resource($this->socket) === true
```

Comment:
Seems like we should maybe move the `is_resource() check` above the `get_resource_type()` check. What do you think? Default the function to return false at the end.

```php
if(is_resource($this->socket) === true) {
    $type = strtolower(get_resource_type($this->socket));
    return ($type == 'socket' || $type == 'stream');
}

return false;
```